src/gateway/http.js

```diff
@@ -54,7 +54,8 @@ HTTP.prototype = Gateway.extends({
 
     const requestParams = assign(defaults, {
       method: requestMethod,
-      headers: assign(headers, this.request.headers())
+      headers: assign(headers, this.request.headers()),
+      timeout
```

Comment:
Since this timeout is different than `request.setTimeout` let's use a configuration to enable it. My recommendation is to use `useSocketConnectionTimeout`. You can add a default value of `false` here: https://github.com/tulios/mappersmith/blob/master/src/mappersmith.js#L73

We should also in the documentation of the new configuration refer to the original nodejs issue (https://github.com/nodejs/node/pull/8101) and explain that if you enable this option it will set a timeout for the entire request, including DNS resolution and socket connection.